### DISCUSSION TITLE
How do I search multiple arbitrary paths on Windows using CMD.exe without exceeding the character limit?

### DISCUSSION THREAD

**Discussion by waivek** [BurntSushi/ripgrep#2307] [Q&A] [answered]
I have approximately 150 files in many different folders (basically all the python files I have edited in vim)

Till now, if I wanted to search them, I'd do `rg <pattern> file_1 file_2 ... file_100` to search them.

This is no longer possible as I am now hitting the CMD.exe character limit of 8192. 

There is no xargs on windows, and I would like to avoid powershell because even that has a character limit of 32k, which is not a permanent solution.

**Comment by BurntSushi** [comment]
I'm pretty sure you can install `xargs` on Windows. That's the route I would suggest.

Failing that, you could try using globs in your shell. I don't know how they work on Windows.

Failing that, you'll have to split your command into multiple commands.

Maybe there are other solutions, but this is more of a Windows question and I don't know much about Windows.

**Comment by waivek** [comment, marked as answer]
Okay got an alternative, using the cross-platform tool [rush](/shenwei356/rush#installation)

`rush -i paths.txt --keep-order "rg --vimgrep pattern {}"  > result.txt`
